### PR TITLE
Use correct URL for YDoc server on Cloud

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Fix https://github.com/enso-org/cloud-v2/issues/1314
  - Use correct URL for YDoc server on Cloud

### Important Notes
- Note that Cloud Projects should also be checked on the desktop app.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~

### CODE REVIEW

File: app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts [hunk 1]
```diff
@@ -28,14 +28,19 @@ const STATUS_SUCCESS_LAST = 299
...
 const STATUS_NOT_AUTHORIZED = 401
 
 /** The number of milliseconds in one day. */
 const ONE_DAY_MS = 86_400_000
 
+/** The internal port of the JSON WebSocket server in a Cloud VM. */
+const JSON_WS_SERVER_PORT = 30001
+/** The `ls` parameter for the YDoc server in a Cloud VM. */
+const YDOC_QUERY_STRING = new URLSearchParams({
+  ls: `ws://localhost:${JSON_WS_SERVER_PORT}`,
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts]: If it's cloud why localhost then?

File: app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts [hunk 2]
```diff
@@ -28,14 +28,19 @@ const STATUS_SUCCESS_LAST = 299
...
- * HTTP status indicating that the request was successful, but the user is not authorized to access
- */
+/** HTTP status indicating that the request was successful, but the user is not authorized to access. */
 const STATUS_NOT_AUTHORIZED = 401
 
 /** The number of milliseconds in one day. */
 const ONE_DAY_MS = 86_400_000
 
+/** The internal port of the JSON WebSocket server in a Cloud VM. */
+const JSON_WS_SERVER_PORT = 30001
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts]: Maybe it's better to put these vars into env?

File: app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts (hunk 1 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts]: the ydoc server is on the same vm as the language server, so from the point of view of the ydoc server the language server is on localhost

**Comment by PabloBuchu** [comment]
So there is not connection error which is good but at the same time nothing is showing up in graph editor. Any way I can help with debugging?

Screenshot 2024-07-16 at 16 49 13


although in logs I can see `failed to authenticate user null`

Screenshot 2024-07-16 at 16 50 01

**Comment by PabloBuchu** [comment]
I see this: 
```
wss.onconnect = (socket, url) => {
  const doc = docName(url.pathname)
  const ls = url.searchParams.get('ls')
  if (doc != null && ls != null) {
    console.log('setupGatewayClient', ls, doc)
    setupGatewayClient(socket, ls, doc)
  } else {
    console.log('Failed to authenticate user', ls, doc)
  }
}

export function docName(pathname: string) {
  const prefix = '/project/'
  if (pathname != null && pathname.startsWith(prefix)) {
    const docName = pathname.slice(prefix.length)
    if (docNameRegex.test(docName)) {
      return docName
    }
  }
  return null
}
``` 
how does the local ydoc url looks like?

File: app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts (hunk 2 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts]: possibly? i don't imagine we'd ever want to change it though, *although* it's definitely better if we didn't have to pass it in at all and instead the cloud should somehow automatically handle everything... one possible solution is to have cloud pass the appropriate query params in its reverse proxy configuration, then frontend doesn't have to do anything. @PabloBuchu thoughts? is this possible? (can `proxy_pass` redirect to `localhost:1234/?ls=ws://localhost:30001`?)

**Review by PabloBuchu — APPROVED** [review]

**Review by MrFlashAccount — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts (hunk 2 above)
Comment (PabloBuchu) [reply app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts]: proxy pass should append any pathname and query params found in url

**Pull request merged**